groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -704,55 +1010,415 @@ struct Optional_Data<TYPE, true> : public Optional_DataImp<TYPE> {
-        d_buffer.address(),
-        allocator,
-        BSLS_COMPILERFEATURES_FORWARD(ARGS, args)...);
-    d_hasValue = true;
-}
+  public:
+    // PUBLIC TYPES
+    typedef TYPE value_type;
+        // 'value_type' is an alias for the underlying 'TYPE' upon which this
+        // template class is instantiated, and represents the type of the
+        // managed object.  The name is chosen so it is compatible with the
+        // 'std::optional' implementation.
+
+    typedef typename bsl::allocator<char> allocator_type;
+
+  private:
+    // PRIVATE TYPES
+    typedef BloombergLP::bslmf::MovableRefUtil MoveUtil;
+
+#ifndef BSLS_COMPILERFEATURES_SUPPORT_OPERATOR_EXPLICIT
+    // UNSPECIFIED BOOL
+
+    // This type is needed only in C++03 mode, where 'explicit' conversion
+    // operators are not supported.  A 'function' is implicitly converted to
+    // 'UnspecifiedBool' when used in 'if' statements, but is not implicitly
+    // convertible to 'bool'.
+    typedef BloombergLP::bsls::UnspecifiedBool<optional> UnspecifiedBoolUtil;
+    typedef typename UnspecifiedBoolUtil::BoolType       UnspecifiedBool;
+
+#endif
+
+    // DATA
+    BloombergLP::bslstl::Optional_Data<TYPE> d_value;
+        // in-place 'TYPE' object
+
+    allocator_type                           d_allocator;
+        // allocator to be used for all in-place 'TYPE' objects
+
+  public:
+    // TRAITS
+    BSLMF_NESTED_TRAIT_DECLARATION(optional,
+                                   BloombergLP::bslma::UsesBslmaAllocator);
+    BSLMF_NESTED_TRAIT_DECLARATION(optional,
+                                   BloombergLP::bslmf::UsesAllocatorArgT);
+    BSLMF_NESTED_TRAIT_DECLARATION_IF(
+        optional,
+        BloombergLP::bslmf::IsBitwiseMoveable,
+        BloombergLP::bslmf::IsBitwiseMoveable<TYPE>::value);
+
+#ifndef BSLS_LIBRARYFEATURES_HAS_CPP11_BASELINE_LIBRARY
+    BSLMF_NESTED_TRAIT_DECLARATION_IF(optional,
+                                      bsl::is_trivially_copyable,
+                                      bsl::is_trivially_copyable<TYPE>::value);
+    // Workaround for C++03 'bsl::is_trivially_copyable' trait.
+#endif  //BSLS_LIBRARYFEATURES_HAS_CPP11_BASELINE_LIBRARY
+
+    // CREATORS
+    optional();
+        // Create a disengaged 'optional' object.  Use the currently installed
+        // default allocator to supply memory for future 'value_type' objects.
+
+    optional(bsl::nullopt_t);                                       // IMPLICIT
+        // Create a disengaged 'optional' object.  Use the currently installed
+        // default allocator to supply memory for future 'value_type' objects.
+
+    optional(const optional& rhs);                                  // IMPLICIT
+        // If 'rhs' contains a value, initialize the contained value using
+        // '*rhs'. Otherwise, create a disengaged 'optional'.  Use the
+        // currently installed default allocator to supply memory for this and
+        // any future 'value_type' objects.
```

Comment:
Please use `original` as the argument name.  In general, the best place to check this type of things is `baltzo_localtimedescriptor` - it's our "master" component.  

Same for other copy/mode constructors

Reply:
changed 'rhs' to 'original' in constructors, and to 'other' in 'swap' and 'value_or'. Left 'rhs' as 'rhs' in assignment operators.

Reply:
modified to 'original' in constructors taking an optional, 'value' in constructors taking a value, 'other' in member functions taking an optional, 'value' in member functions taking a value.